static/app/components/profiling/flamegraphOptionsMenu.tsx

```diff
@@ -17,6 +17,42 @@ function FlamegraphOptionsMenu({
 }: FlamegraphOptionsMenuProps): React.ReactElement {
   const [{colorCoding, xAxis}, dispatch] = useFlamegraphPreferences();
 
+  const options = useMemo(() => {
+    return [
+      {
+        label: t('X Axis'),
+        value: 'x axis',
+        defaultValue: xAxis,
+        options: Object.entries(X_AXIS).map(([value, label]) => ({
+          label,
+          value,
+        })),
+        onChange: value =>
+          dispatch({
+            type: 'set xAxis',
+            payload: value,
+          }),
+      },
+      {
+        label: t('Color Coding'),
+        value: 'by symbol name',
+        defaultValue: colorCoding,
+        options: Object.entries(COLOR_CODINGS).map(([value, label]) => ({
+          label,
+          value,
+        })),
+        onChange: value =>
+          dispatch({
+            type: 'set color coding',
+            payload: value,
+          }),
+      },
+    ];
+    // If we add color and xAxis it updates the memo and the component is re-rendered (losing hovered state)
```

Comment:
@vuluongj20 this works for us because we dont rely on any of the computed properties inside the options, but is worth noting that it would break - I think persisting the hovered node could be an option that we should explore while the component is open